packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -448,20 +406,35 @@ contract ControllerHelper is UniswapControllerHelper, EulerControllerHelper, IER
         ControllerHelperDataType.RebalanceVaultNftParams[] calldata _params
     ) external payable {
         // check ownership
-        require(IShortPowerPerp(ControllerHelperDiamondStorage.getAddressAtSlot(2)).ownerOf(_vaultId) == msg.sender);
+        _checkAccess(_vaultId);
 
         wrapInternal(msg.value);
         _flashLoan(
-            ControllerHelperDiamondStorage.getAddressAtSlot(5),
+            weth,
             _collateralToFlashloan,
             uint8(ControllerHelperDataType.CALLBACK_SOURCE.FLASHLOAN_REBALANCE_VAULT_NFT),
             abi.encode(_vaultId, _params)
         );
 
-        ControllerHelperUtil.sendBack(
-            ControllerHelperDiamondStorage.getAddressAtSlot(5),
-            ControllerHelperDiamondStorage.getAddressAtSlot(4)
-        );
+        ControllerHelperUtil.sendBack(weth, wPowerPerp);
+    }
+
+    /**
+     * @notice checks if the msg.sender is the owener of the vault
+     * @param _vaultId vault ID
+     */
+    function _checkAccess(uint256 _vaultId) internal view {
+        require(IShortPowerPerp(shortPowerPerp).ownerOf(_vaultId) == msg.sender);
+    }
+
+    /**
+     * @notice approve controller to transfer uniswap nft and deposit in vault
+     * @param _vaultId vault ID
+     * @param _tokenId uniswap nft token id
+     */
+    function _approveDepositNft(uint256 _vaultId, uint256 _tokenId) internal {
+        INonfungiblePositionManager(nonfungiblePositionManager).approve(controller, _tokenId);
```

Comment:
I think this is not needed as we approve all in the constructor, right?

Reply:
gah, I removed it and then added it back in